src/generateDocumentation.js

```diff
@@ -292,12 +292,33 @@ function generateModels(templateModels) {
 
   templateModels.forEach((templateModel) => {
     if (!templateModel.name || !templateModel.schema) return;
-    models[templateModel.name] = templateModel.schema;
+    models[templateModel.name] = generateModelSchema(templateModel.schema);
   });
 
   return models;
 }
 
+function generateModelSchema(templateSchema) {
+  if (!templateSchema) return templateSchema;
+
+  let schema = {};
+  Object.keys(templateSchema).forEach((key) => {
+    let field = templateSchema[key];
+    if (key === "$ref") {
+      const match = /{{\s*model\s*:\s*([\-\w]+)\s*}}/.exec(field);
+      if (match) {
+        field = `#/components/schemas/${match[1]}`;
+      }
+    } else if (!Array.isArray(field) && typeof field === "object" && field) {
```

Comment:
If I understand the code well this mind be simpler:
```
} else if (field.constructor === Object && field) {
```
or
```
} else if (key === "items" || key === "properties") {
```

Reply:
You're totally right on using `field.constructor === Object`. Changed!

Regarding checking directly the value of `key`, which to be honest was my first attempt, the issue is that the _"properties"_ field is made of other inner objects (e.g., _"properties.[SUBFIELD].$ref"_ or _"properties.[SUBFIELD].items.$ref"_, but potentially also  _"properties.[SUBFIELD].properties.[SUB-SUBFIELD].$ref"_). Thus, when we arrive to this check, the value of `key` is already different than _"properties"_.

And, in the end, I have opted for simplicity and readability over performance.

Reply:
Ok